### ISSUE TITLE
Specifying --env-file returns a UID / GID readonly warning

### ISSUE THREAD

**Issue by jmmaljaars** [sylabs/singularity#1263]
**Version of Singularity**

```
singularity-ce version 3.10.2-bionic
```

**Describe the bug**

Whenever I attempt to pass environment variables to a singularity container using the `--env-file` option, the following warning is issued:

```
GID: readonly variable
UID: readonly variable
```
even if the `--env-file` is actually empty.

**To Reproduce**

1. create a file containing the environment variables as key-value pairs, see https://docs.sylabs.io/guides/3.7/user-guide/environment_and_metadata.html#env-file-option , e.g.

```
echo MY_VAR=host_var > .env
```

2. get an interactive singularity session, e.g.

```
singularity shell --env-file .env $SINGULARITY_CONTAINER
```

in which the $SINGULARITY_CONTAINER points to a valid .sif file.

Results in:

```
GID: readonly variable
UID: readonly variable
Singularity> echo $MY_VAR 
host_var
```

i.e. the variable is passed correctly, but I get an annoying GID / UID warning.

**Expected behavior**
I didn't expect the UID/GID readonly variable warnings.

**OS / Linux Distribution**
Which Linux distribution are you using?
```
Ubuntu 18.04.6 LTS
```

**Installation Method**
Installed by sysadmin, not sure which method he followed.

**Comment by dtrudg** [comment]
I'm afraid that this is not a bug, and is expected behaviour.

Specific environment variables are set readonly by the shell. You cannot modify them directly. Because singularity uses a shell interpreter to handle environment variables, the warnings you see are expected.

The same can be seen if you try to set `UID` or `GID` directly on your host's bash shell:

```
03:45 pm $ UID=abc
bash: UID: readonly variable
```

**Comment by dtrudg** [comment]
Okay - sorry... I replied to quickly there, with a misunderstanding. This does appear to be a bug.

**Comment by jmmaljaars** [comment]
> I'm afraid that this is not a bug, and is expected behaviour.
> 
> Specific environment variables are set readonly by the shell. You cannot modify them directly. Because singularity uses a shell interpreter to handle environment variables, the warnings you see are expected.
> 
> The same can be seen if you try to set `UID` or `GID` directly on your host's bash shell:
> 
> ```
> 03:45 pm $ UID=abc
> bash: UID: readonly variable
> ```
> 
> edit - actually maybe I am confused here... you are getting the warning with _any_ `--env-file`, not just ones containing UID / GID? What `SINGULARITYENV_` variables are set?

Indeed, get this warning with `--env-file`, even if the file itself is actually empty / trivial (i.e. definitely not containing UID, or GID as keys). Also, 

```
printenv | grep SINGULARITYENV
```

is empty.

**Comment by dtrudg** [comment]
@jmmaljaars - this has been fixed in the `main` branch, and will be in the future 3.11 release. Thanks for the report.
